browser/src/Utility.ts

```diff
@@ -246,9 +246,13 @@ export function ignoreWhilePendingPromise<T, U>(
     return ret
 }
 
-export function checkIfFileExistsSync(filename: string): boolean | Error {
+export function checkIfPathExists(
+    fileOrFolder: string,
+    type: "file" | "folder" = "file",
+): boolean | Error {
     try {
-        return fs.statSync(filename).isFile()
+        const stats = fs.statSync(fileOrFolder)
```

Comment:
Awww sync code :cry:

Reply:
@DeltaEvo my changes here are a work in progress, I'm switching things over to use the async methods since your PR just working on resolving the issues with tests, once I push local changes that call will be async

Reply:
:heart: